src/common/io.ts

```diff
@@ -166,12 +162,6 @@ export class ARIOReadable implements AoARIORead {
   async getEpochSettings(params?: EpochInput): Promise<AoEpochSettings> {
     const allTags = [
       { name: 'Action', value: 'Epoch-Settings' },
-      {
-        name: 'Timestamp',
-        value:
-          (params as { timestamp?: number })?.timestamp?.toString() ??
-          (await getCurrentBlockUnixTimestampMs(this.arweave)).toString(),
-      },
```

Comment:
after inspection, it looks like CUs set the timestamp based on the last message evaluation they've processed. that _should_ be sufficient for what we need, but will require some more testing. this would allow us to drop the use of `TImestamp` tags everywhere in the contract.